admiral/envs/components/state.py

```diff
@@ -0,0 +1,337 @@
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from admiral.envs.components.agent import LifeAgent, PositionAgent, SpeedAngleAgent
+
+# ----------------------- #
+# --- Health and Life --- #
+# ----------------------- #
+
+class LifeState:
+    """
+    Agents can die if their health falls below their minimal health value. Health
+    can decrease in a number of interactions. This environment provides an entropy
+    that indicates how much health an agent loses when apply_entropy is called.
```

Comment:
This is a generic entropy for the step. If you want to specify health penalties for specific actions, such as being attacked, you must write it in the environment.